chromium_src/ios/web/webui/mojo_facade.mm

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "src/ios/web/webui/mojo_facade.mm"
+
+namespace web {
+bool MojoFacade::IsWebUIMessageAllowedForFrame(WKFrameInfo* frame,
+                                               const GURL& origin,
+                                               NSString** prompt) {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  CHECK(prompt && *prompt);
+
+  auto name_and_args =
+      GetMessageNameAndArguments(base::SysNSStringToUTF8(*prompt));
+
+  // If the scheme is untrusted
+  if (name_and_args.name == "Mojo.bindInterface" &&
+      origin.scheme() == "chrome-untrusted") {
+    const base::Value::Dict& args = name_and_args.args;
+    const std::string* interface_name = args.FindString("interfaceName");
+    CHECK(interface_name);
+
+    // Check if the requested interface is registered for this origin
+    bool is_allowed =
+        web_state_->GetInterfaceBinderForMainFrame()->IsAllowedForOrigin(
+            origin, *interface_name);
+
+    // If the interface is not allowed, set the prompt to invalid.
+    // HandleMojoMessage will not bind it.
+    if (!is_allowed) {
+      *prompt = @"{\"name\":\"Mojo.invalidInterface\",\"args\":{}}";
```

Comment:
I think we should just be returning `false` here which would be the same as a normal page trying to access a mojo interface.

Reply:
I wasn't sure what to do here because if I return `false` it could show a `popup` on the WebView because it falls through to `shouldPresentJavaScriptDialogForRequestURL` https://source.chromium.org/chromium/chromium/src/+/main:ios/web/web_state/ui/crw_wk_ui_handler.mm;l=266

Which `will` show a popup (erroneously imo): https://source.chromium.org/chromium/chromium/src/+/main:ios/web/web_state/ui/crw_wk_ui_handler.mm;l=336 for non-main frame requests (usually untrusted isn't the main-frame and is embedded in a `chrome://` afaik).

Reply:
I think showing a popup is fine here because this should never happen unless some malicious code is trying to take advantage of it. The popup will make that more obvious and I think in general it should behave the same as the trusted version if the interface isn't registered

Reply:
Fixed: https://github.com/brave/brave-core/pull/26402/commits/86be840e64541b3df0e092c80591843cf1a7e19c